src/msg.cpp

```diff
@@ -169,6 +169,7 @@ int zmq::msg_t::init_data (void *data_,
         _u.lmsg.content =
           static_cast<content_t *> (malloc (sizeof (content_t)));
         if (!_u.lmsg.content) {
+            ffn_(data_,hint_);
```

Comment:
In case of error we leave the user data alone - it's their responsibility to take care of it - or in most cases, to simply try again without redoing it

Reply:
More importantly, this would be an API change so we can't really merge as-is, sorry, needs to be fixed first

Reply:
@TomMD ping